ring-core/src/ring/util/response.clj

```diff
@@ -92,13 +92,22 @@
       (.exists file)
         file)))
 
-(defn- file-content-length [resp]
-  (let [file ^File (:body resp)]
-    (header resp "Content-Length" (.length file))))
+(defn- file-info [^File file]
+  (if file
+    {:content        file
+     :content-length (.length file)
+     :directory?     (.isDirectory file)
+     :last-modified  (last-modified-date file)}))
 
-(defn- file-last-modified [resp]
-  (let [file ^File (:body resp)]
-    (header resp "Last-Modified" (format-date (last-modified-date file)))))
+(defn- content-length [resp len]
+  (if len
```

Comment:
This was to allow implementations of resource-info to return nil values, but don't think allowing nil values is a good idea.  I'm going to change the docstring to state that all of the values (except :content) must me non-nil, and follow the URLConnection convention for unknown content-length and last-modified values (-1 and 0, respectively).